pkg/sql/plan2/make.go

```diff
@@ -205,3 +207,19 @@ func MakePlan2DefaultExpr(expr engine.DefaultExpr) *plan.DefaultExpr {
 	}
 	return ret
 }
+
+// if typ is decimal128 and decimal64 without scalar and precision
+// set a default value for it.
+func rewriteDecimalTypeIfNecessary(typ *plan.Type) *plan.Type {
+	if typ.Id == plan.Type_DECIMAL128 && typ.Scale == 0 && typ.Width == 0 {
+		typ.Scale = 10
+		typ.Width = 38
+		typ.Size = 16
+	}
+	if typ.Id == plan.Type_DECIMAL64 && typ.Scale == 0 && typ.Width == 0 {
+		typ.Scale = 2
+		typ.Width = 6
+		typ.Size = 16
```

Comment:
Size = 8?

Reply:
oh, I made a mistake when I copy the code. Sorry. It's 8.